doc/overlays.xml

```diff
@@ -96,4 +96,68 @@ overridden and/or new packages.</para>
 
 </section>
 
+<section xml:id="sec-overlays-extend-modify">
+<title>Overlays used to extend or modify pkgs</title>
+
+<para>
+If you want to extend nixpkgs with your own packages: add new packages, alter existing ones then you can use this code <filename>nixcloud-pkgs.nix</filename>:
+</para>
+
+<programlisting>
+{  pkgs ? import &lt;nixpkgs> {}
+,  nixcloud-backend   ? pkgs.stdenv.lib.cleanSource ../nixcloud-backend
+,  nixcloud-frontend  ? pkgs.stdenv.lib.cleanSource ../nixcloud-frontend
+,  nixcloud-editor    ? pkgs.stdenv.lib.cleanSource ../nixcloud-editor
+, ...
+}:
+let
+  nc-backend = nixcloud-backend;
+  nc-frontend = nixcloud-frontend;
+  
+  newpkgs = import pkgs.path { overlays = [ (pkgsself: pkgssuper: {
+    nixcloud-backend  = pkgs.callPackage nc-backend { inherit newpkgs; };
+    nixcloud-frontend = pkgs.callPackage "${nc-frontend}/release.nix" {
+      inherit nixcloud-editor;
+    };
+  } ) ]; };
+in newpkgs
+</programlisting>
```

Comment:
This snippet is not the recommended way to make overlays, because it is not modular.  Also it is incorrect because it mention `nixpkgs` as one input.  Your overlay should probably look like that instead:

```nix
self: super:

let
  path = if builtins.pathExists <nixcloud> then <nixcloud> else ../.;
in  

{
  nixcloud-backend  = super.callPackage (super.lib.cleanSource "${path}/nixcloud-backend") { };
  nixcloud-frontend = super.callPackage "${path}/nixcloud-frontend/release.nix" {
    nixcloud-editor = super.lib.cleanSource "${path}/nixcloud-editor";
  };
}
```

And be installed in the `~/.config/nixpkgs/overlays` directory with a symbolic link, such that the `path` resolution can use a relative path.